compiler/rustc_span/src/lib.rs

```diff
@@ -2149,3 +2149,17 @@ where
         Hash::hash(&len, hasher);
     }
 }
+
+/// Useful type to use with `Result<>` indicate that an error has already
```

Comment:
As I noted in the PR description, this shouldn't live here -- it's just here as a proof of concept for the rest of the PR.

Where should this live instead? :thinking: